plugin/src/main/java/org/opensearch/ml/action/deploy/TransportDeployModelAction.java

```diff
@@ -133,26 +138,33 @@ public TransportDeployModelAction(
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLDeployModelResponse> listener) {
         MLDeployModelRequest deployModelRequest = MLDeployModelRequest.fromActionRequest(request);
         String modelId = deployModelRequest.getModelId();
-        Boolean isUserInitiatedDeployRequest = deployModelRequest.isUserInitiatedDeployRequest();
+        String tenantId = deployModelRequest.getTenantId();
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, tenantId, listener)) {
+            return;
+        }
+        boolean isUserInitiatedDeployRequest = deployModelRequest.isUserInitiatedDeployRequest();
         User user = RestActionUtils.getUserContext(client);
         boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
         String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<MLDeployModelResponse> wrappedListener = ActionListener.runBefore(listener, () -> context.restore());
-            mlModelManager.getModel(modelId, null, excludes, ActionListener.wrap(mlModel -> {
+            ActionListener<MLDeployModelResponse> wrappedListener = ActionListener.runBefore(listener, context::restore);
+            mlModelManager.getModel(modelId, tenantId, null, excludes, ActionListener.wrap(mlModel -> {
                 FunctionName functionName = mlModel.getAlgorithm();
                 Boolean isHidden = mlModel.getIsHidden();
+                if (!TenantAwareHelper.validateTenantResource(mlFeatureEnabledSetting, tenantId, mlModel.getTenantId(), listener)) {
```

Comment:
Less a comment for this PR and more one to think about for later.  This returns 401 (Unauthorized) but we might want to consider returning 404 (Not Found) in many cases when trying to manipulate a tenant resource that you don't own.

Reply:
Yeah that can be an option. We can decide later and address that in a separate PR.